test/suggestion.cpp

```diff
@@ -0,0 +1,188 @@
+    search.set_range(0, range);
+    search.set_verbose(true);
+    
+    std::vector<std::string> result;
+    for (auto entry = search.begin();entry!=search.end();entry++) {
+      result.push_back((*entry).getTitle());
+    }
+    
+    return result;
+  }
+
+  TEST(Suggestion, emptyQuery) {
+    std::vector<std::string> titles = {
+                                        "fooland",
+                                        "berlin wall", 
+                                        "hotel berlin, berlin", 
+                                        "again berlin", 
+                                        "berlin", 
+                                        "not berlin", 
+                                       };
+ 
+    std::string path = createTestZim(titles);
+    const zim::Archive archive(path);
+    std::vector<std::string> resultSet = getSuggestions(archive, "", archive.getEntryCount());
+    
+    ASSERT_EQ(resultSet.size(), 0);
+  }
+
+  TEST(Suggestion, noResult) {
+    std::vector<std::string> titles = {
+                                        "fooland"
+                                        "berlin wall", 
+                                        "hotel berlin, berlin", 
+                                        "again berlin", 
+                                        "berlin", 
+                                        "not berlin", 
+                                       };
+ 
+    std::string path = createTestZim(titles);
+    const zim::Archive archive(path);
+    std::vector<std::string> resultSet = getSuggestions(archive, "none", archive.getEntryCount());
+    
+    ASSERT_EQ(resultSet.size(), 0);
+  }
+
+  TEST(Suggestion, singleTermOrder) {
+    std::vector<std::string> titles = {
+                                        "fooland",
+                                        "berlin wall", 
+                                        "hotel berlin, berlin", 
+                                        "again berlin", 
+                                        "berlin", 
+                                        "not berlin", 
+                                       };
+    
+    std::vector<std::string> expectedResult = {
+                                        "berlin", 
+                                        "hotel berlin, berlin", 
+                                        "not berlin", 
+                                        "berlin wall", 
+                                        "again berlin", 
+                                       };
+ 
+    std::string path = createTestZim(titles);
+    const zim::Archive archive(path);
+    std::vector<std::string> resultSet = getSuggestions(archive, "berlin", archive.getEntryCount());
+    
+    ASSERT_EQ(expectedResult.size() , resultSet.size());
+    
+    for(int i = 0;(unsigned int)i < resultSet.size();i++){
+      std::cout<<"Expected: "<<expectedResult[i]<<", Actual: "<<resultSet[i]<<std::endl;
+      ASSERT_EQ(expectedResult[i], resultSet[i]);
+    }
```

Comment:
Can't this be replaced with `ASSERT_EQ(expectedResult, resultSet)`?

Reply:
@veloman-yunkan Do we have a class/solution which allow to handle these tmp_file elegantly (without having to remove the file explecitly for example)?